### PR TITLE
Loads of lint fixes. Removing unused/uneeded code.

### PR DESCRIPTION
**Pull request by canndrew**
This change is [Reviewable](https://reviewable.io/reviews/maidsafe/crust/620)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @vinipsmaker

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/tcp_connections.rs [hunk 1]
```diff
@@ -71,50 +70,6 @@ fn upgrade_writer(mut stream: TcpStream) -> Sender<WriteEvent> {
     tx
 }
 
-/// Returns the stream along with the peer's SocketAddr
```
Comment (dirvine) [inline src/tcp_connections.rs]: Is this functionality elsewhere now ?

Comment (canndrew) [inline src/tcp_connections.rs]: Yes. Hole punching is implemented in the nat_traversal crate.

Comment (dirvine) [inline src/tcp_connections.rs]: :+1:

**Comment by vinipsmaker** [comment]
Reviewed 11 of 15 files at r1, 4 of 4 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/crust/620)*

**Pull request merged**